c20_game_of_life/game_of_life.go

```diff
@@ -0,0 +1,137 @@
+package main
+
+import (
+	"fmt"
+	"math/rand"
+	"time"
+)
+
+type universe [][]bool
+
+// Creates a new instance of a universe - every cell is initialized to false
+func newUniverse(height int, width int) universe {
+	h := make([][]bool, height)
+	for w, _ := range h {
+		h[w] = make([]bool, width)
+	}
+	return h
+}
+
+// prints out a universe's 2d array, using 1 for true and 0 for false
+func (u universe) showUniverse() {
+	c0 := "0"
+	c1 := "1"
+	for _, h := range u {
+		for _, w := range h {
+			if w {
+				fmt.Printf(" %v", c1)	
+			} else {
+				fmt.Printf(" %v", c0)
+			}
+		}
+		fmt.Printf("\n")
+	}
+}
+
+// calculates the percentage of cells that are currently alive
+func (u universe) calcPopPercent() float64 {
+	var totalCells int
+	var liveCells int
+	for _, h := range u	{
+		totalCells += len(h)
+		for w, _ := range h {
+			if h[w] == true {
+				liveCells++
+			}
+		}
+	}
+	return float64(liveCells) / float64(totalCells) * 100
+}
+
+// prints a universe's state and percentage of living cells
+func (u universe) printUniverseInfo() {
+	u.showUniverse()
+	fmt.Printf("living cell percentage: %v%%\n", u.calcPopPercent())
+}
+
+// seeds a universe, with a 25% chance for each cell to become alive
+func (u universe) seedUniverse() {
+	rand.Seed(time.Now().UnixNano())
+	for _, h := range u	{
+		for w, _ := range h {
+			h[w] = rand.Intn(4) == 3
+		}
+	}
+}
+
+// If x or y are out of range, it loops the coordinate to the other side
+func (u universe) treatCoord(y int, x int) (int, int) {
+	if y >= len(u) {
+		y = y % len(u)
+	} else if y < 0 {
+		y = (y % -len(u)) + len(u)
+	}
+	if x >= len(u[y]) {
+		x = x % len(u[y])
+	} else if x < 0 {
+		x = (x % -len(u[y])) + len(u[y])
+	}
+	return y, x
+}
+
+// tests a cell to see if it is alive (set to true)
+func (u universe) isAlive(y int, x int) bool {
+	return u[y][x]
+}
+
+// calculates how many cells around a coordinate has a living cell
+func (u universe) neighbors(y int, x int) int {
+	var n int = 0
+	var ty, tx int
+	for h := -1; h < 2; h++ {
+		for w := -1; w < 2; w++ {
+			if h != 0 || w != 0 {
+				ty, tx = u.treatCoord(y + h, x + w)
+				if u[ty][tx] {
+					n++
+				}
+			}
+		}
+	}
+	return n
+}
+
+// calculates if a cell will continue to live, or if a cell will start living
+func (u universe) willLive(y int, x int) bool {
```

Comment:
Using `y` and `x` like this is strange to me. I think it's more clear to use `h` and `w`, or even `i` and `j` as those variables often represent integers.